src/store/ChartAdapterStore.ts

```diff
@@ -237,38 +247,85 @@ export default class ChartAdapterStore {
+                if (e.type === 'touchend') {
+                    if (!this.touchValues.touchIds?.length) {
+                        this.touchValues = { multiTouch: false };
+                    } else {
+                        this.touchValues.touchIds = this.touchValues.touchIds?.filter(
+                            id => id === e.changedTouches[0].identifier
+                        );
+                    }
+                }
+                return;
+            }
+            if (e.touches.length > 1) {
+                this.touchValues = { multiTouch: true };
+                this.touchValues.touchIds = Array.from(e.touches).map(t => t.identifier);
+                return;
+            }
+
+            const { pageX, pageY } = e.changedTouches[0];
+
+            if (['touchmove', 'touchend'].includes(e.type)) {
                 const nonScrollableAreaWidth = chartNode.offsetWidth - this.mainStore.chart.yAxisWidth;
                 const { left } = chartNode.getBoundingClientRect();
+                const x = pageX - left;
+                const isForcedScrollArea = x < nonScrollableAreaWidth;
 
                 if (this.touchValues.x && this.touchValues.y) {
-                    const deltaX = Math.abs(screenX - this.touchValues.x);
-                    const deltaY = Math.abs(screenY - this.touchValues.y);
+                    const xDiff = this.touchValues.x - pageX;
+                    const yDiff = this.touchValues.y - pageY;
+                    const deltaXTotal = (this.touchValues.deltaXTotal ?? 0) + xDiff;
+                    const deltaYTotal = (this.touchValues.deltaYTotal ?? 0) + yDiff;
+                    const deltaX = e.type === 'touchend' ? Math.abs(deltaXTotal) : Math.abs(xDiff);
+                    const deltaY = e.type === 'touchend' ? Math.abs(deltaYTotal) : Math.abs(yDiff);
                     const isVerticalScroll = deltaY > deltaX;
-                    const x = pageX - left;
-                    if (x < nonScrollableAreaWidth && isVerticalScroll && !this.scrollChartParentOnTouchTimer) {
-                        this.touchValues.yOnTouchEnd = undefined;
-                        this.scrollChartParentOnTouchTimer = setTimeout(() => {
-                            this.scrollableChartParent?.scrollBy({
-                                top: screenY - Number(this.touchValues.yOnTouchEnd ?? this.touchValues.y),
+                    this.touchValues = { ...this.touchValues, deltaXTotal, deltaYTotal };
+
+                    if (isForcedScrollArea && isVerticalScroll) {
+                        const shouldForceMaxScroll =
+                            Math.abs(Number(this.touchValues.deltaYTotal)) > 10 && e.type === 'touchend';
+                        const stopChartPagination = () => {
+                            // calling _scrollAnimationController.stop() in flutter-chart, value 1 doesn't affect the scale:
+                            this.flutterChart?.app.scale(1);
```

Comment:
this triggers the following method in flutter-chart which allows to stop chart pagination when we are scrolling:
<img width="575" alt="Screenshot 2024-08-13 at 5 08 11 PM" src="https://github.com/user-attachments/assets/d67b51fb-0b05-4f7d-90b4-c9afcdb494c5">